main.py

```diff
@@ -95,6 +96,30 @@ async def get_experiment_info(file: str) -> Any:
     return remote.examine(file)
 
 
+previous_queue = {}
+current_queue = {}
```

Comment:
Do we need both dictionaries? I think 1 is enough to store the previous state. The current state can be handled locally I guess.